src/NuGet.Core/NuGet.Protocol/Resources/VulnerabilityInfoResourceV3.cs

```diff
@@ -0,0 +1,299 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using NuGet.Common;
+using NuGet.Protocol.Core.Types;
+using NuGet.Protocol.Model;
+
+namespace NuGet.Protocol.Resources
+{
+    /// <summary>Implementation of <see cref="IVulnerabilityInfoResource"/> for NuGet V3 HTTP feeds.</summary>
+    /// <remarks>Not intended to be created directly. Use <see cref="SourceRepository.GetResourceAsync{T}(CancellationToken)"/>
+    /// with <see cref="IVulnerabilityInfoResource"/> for T, and typecast to this class.
+    /// <para>Implements the <a href="https://learn.microsoft.com/nuget/api/vulnerability-info-resource">VulnerabilityInfo server API resource</a>.</para></remarks>
+    public sealed class VulnerabilityInfoResourceV3 : IVulnerabilityInfoResource
+    {
+        private readonly SourceRepository _sourceRepository;
+
+        internal VulnerabilityInfoResourceV3(SourceRepository sourceRepository)
+        {
+            _sourceRepository = sourceRepository ?? throw new ArgumentNullException(nameof(sourceRepository));
+        }
+
+        /// <summary>Get the vulnerability pages the server contains.</summary>
+        /// <param name="cacheContext">The cache settings to use when making HTTP requests.</param>
+        /// <param name="log">The logger for any messages.</param>
+        /// <param name="cancellationToken">The cancellation token to cancel operation.</param>
+        /// <returns>The list of vulnerability data pages the server claims to have.</returns>
+        /// <exception cref="FatalProtocolException">When various HTTP or deserialization exceptions occur.</exception>
+        public async Task<IReadOnlyList<V3VulnerabilityIndexEntry>> GetVulnerabilityFilesAsync(SourceCacheContext cacheContext, ILogger log, CancellationToken cancellationToken)
```

Comment:
I'm guessing this being public is a consequence of some of the earlier discussions?

Reply:
yes: https://github.com/NuGet/NuGet.Client/pull/5115#discussion_r1155728138

Basically, people who maintain their own servers, or server implementations, can use these APIs to mirror another source's vulnerability data in their own instance.